Rebalance reinforcements, plus related tweaks

jaj22 opened the pull request:
### What type of PR is this.
1. [X] Bug
2. [X] Change
3. [ ] Enhancement

### What have you changed and why?
- Balance reinforcement system for player count.
- Enable reinforcing from "carriers".
- Separate road patrol generation from reinforcements and rebalance.
- Sanitize garrison sizes (fewer giant and tiny garrisons, units of 4 rather than 8).
- Use more 4-man teams in garrisons for both init and reinf.
- Prevent new reinf convoys spamming after a recapture.
- Fill out the AA & AT squads with a fourth soldier, cap militia squads to 8.

The rebalanced reinforcements generate an average of around 16 troops per 10 minute tick for nine players at war level 3, or about half that for a single player. For comparison, the previous version generated 8 troops per airport (not carrier) in range. The new version is not particularly dependent on airfield count or positioning.

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Checking max garrison sizes:
`outposts apply { [_x, [_x] call A3A_fnc_garrisonSize] };`

Checking reinforcements:
`[] spawn A3A_fnc_reinforcementsAI`
Check the RPT after each call, there is logging for the total troops and for each reinforcement sent.

HakonRydland commented on A3-Antistasi/functions/CREATE/fn_garrisonSize.sqf:
```diff
@@ -1,35 +1,33 @@
-	_nVeh = _nVeh + 1;
-	}
+{
+    _groups = 2 + round (_size/30);
+    _groups = _groups min 11;
```
can combine the min

jaj22 replied:
Inclined to leave it because it's more readable alongside the other two cases where it can't be combined.

HakonRydland commented on A3-Antistasi/functions/CREATE/fn_reinforcementsAI.sqf:
```diff
@@ -27,84 +27,87 @@ while {killZoneRemove >= 1} do
+
+	// Don't reinforce if marker has enemy-controlled airfields within spawn distance
+	private _siteSide = sidesX getVariable [_site, sideUnknown];
+	if ({(getMarkerPos _x distance2d getMarkerPos _site < distanceSPWN) and (sidesX getVariable [_x,sideUnknown] != _siteSide)} count airportsX > 0) then { continue };
```
findIf would be faster, albeit only slightly

jaj22 replied:
Done. You might want to double-check that one because it's a pain to test.

HakonRydland commented on A3-Antistasi/functions/init/fn_initGarrisons.sqf:
```diff
@@ -57,36 +57,34 @@ _fnc_initMarker =
+		}
+		else
+		{
+			if(_type != "Airport" && {_type != "Outpost"}) then
```
would be better
```sqf
if !(_type in ["Airport", "Outpost"]) then
```

jaj22 replied:
Done.

HakonRydland commented on A3-Antistasi/functions/CREATE/fn_reinforcementsAI.sqf:
```diff
@@ -27,84 +27,87 @@ while {killZoneRemove >= 1} do
-_airportsX = airportsX select {(sidesX getVariable [_x,sideUnknown] != teamPlayer) and (spawner getVariable _x == 2)};
-if (count _airportsX == 0) exitWith {};
-_reinfPlaces = [];
+private _playerScale = (8 + count (allPlayers - entities "HeadlessClient_F")) / 17;
```
again would probably be better as a function

jaj22 replied:
Ok, I guess it won't hurt. For the sake of confusing git, I think what I'll do is switch it to a function in 2124 where it's used repeatedly, get Bob to merge that in first, and then update this one.

HakonRydland reviewed (changes requested):
couple of small things

jaj22 commented:
Updated according to review. Also added an optional parameter to garrisonSize to ignore the frontier status of an marker, because I'm going to need that shortly.

Instructions for @Bob-Murphy: Merge #2124 before this and then ping me so that I can update this PR for the playerScale function separation.

HakonRydland reviewed (commented):
once the player scale function is in use this should be good

jaj22 commented:
Switched over to the getPlayerScale function, should be good to go.

Bob-Murphy commented:
One change request is still open, even tho it's outdated.
Needs to be resolved.
